docs/source/control_tutorials.rst

```diff
@@ -355,26 +364,169 @@ As with :py:class:`.MachineController`, :py:class:`.BMPController` supports the
     ...     if bc.read_adc().temp_top > 75.0:
     ...         bc.set_led(7, True)  # Turn on LED 7 on the board
 
-Using SDP and SCP Directly (Advanced)
--------------------------------------
 
-SCP and SDP packets can be unpacked from strings of :py:class:`bytes` received
-over the network or assembled into bytes for transmission by the following two
-interfaces:
+.. _scp-and-sdp-tutorial:
+
+Sending/receiving SDP and SCP packets to/from applications
+----------------------------------------------------------
+
+A number of low-level facilities are provided for users who wish to send and
+receive SCP and SDP packets directly. The most common use for these APIs is to
+send and receive SDP packets to and from a running SpiNNaker application to
+allow realtime monitoring and communication with the underlying application via
+an IP Tag. A minimal example of each is presented below.
+
+Example: Sending SDP packets to a running application
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In your SpiNNaker application you should register a callback handler for the
+arrival of SDP packets. For example, using the ``spin1_api``:
+
+.. code-block:: c
+
+    spin1_callback_on(SDP_PACKET_RX, on_sdp_from_host, 0);
+
+To send SDP packets to this application, you must open a UDP socket with which
+to send SDP packets to your SpiNNaker system. Note that (slightly confusingly)
+SpiNNaker listens for incoming SDP packets on the :py:data:`SCP port
+<rig.machine_control.consts.SCP_PORT>`.
+
+::
+
+    >>> import socket
+    >>> from rig.machine_control.consts import SCP_PORT
+    >>> out_sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    >>> out_sock.connect((hostname, SCP_PORT))
 
-* :py:class:`~.packets.SDPPacket`
-* :py:class:`~.packets.SCPPacket`
+With the port opened, you can use the
+:py:class:`rig.machine_control.packets.SDPPacket` and
+:py:class:`rig.machine_control.packets.SCPPacket` classes to pack your data
+into properly formatted SDP or SCP packets. Since the ``sark`` and
+``spin1_api`` (unfortunately) make packing/unpacking SDP packets rather clumsy,
+it is common to send and receive SCP packets to applications. 
 
-A blocking implementation of SCP is provided by
-:py:class:`~.scp_connection.SCPConnection`.
+.. note::
 
-These are used internally by :py:class:`.MachineController` and
-:py:class:`.BMPController`. Users are encouraged to read the official SDP and
-SCP App-Notes and refer to the Rig source code for further guidance in using SCP
-and SDP directly in applications.
+    SCP packets are just SDP packets with some additional fields placed in the
+    SDP data payload and when a port number other than 0 are just passed to
+    your application like any other SDP packets.
```

Comment:
"payload.  When a port number other than 0 is used SCP packets are passed to the application like any other SDP packet."